app/components/login_button_component.rb

```diff
@@ -0,0 +1,30 @@
+class LoginButtonComponent < ButtonComponent
+  APP_NAME = 'Login.gov'
+  VALID_COLORS = ['primary', 'primary-darker', 'primary-lighter'].freeze
+
+  attr_reader :color, :tag_options
+
+  def initialize(color: 'primary', **tag_options)
+    if !VALID_COLORS.include?(color)
+      raise ArgumentError, "`color` #{color}} is invalid, expected one of #{VALID_COLORS}"
+    end
+
+    super(
+      color: color,
+      **tag_options
+    )
+
+    @color = color
+  end
+
+  def logo_class
+    return 'login-button__logo login-button__logo-white' if color == 'primary-darker'
+    'login-button__logo'
+  end
+
+  def css_class
+    classes = super || ['usa-button', *tag_options[:class]]
```

Comment:
I'm not sure I follow if / how this class is extended to be a subclass of `ButtonComponent`, where if it were a subclass, I'd expect we wouldn't duplicate `usa-button` here, and we'd be deferring the rendering to `ButtonComponent` in some way.

I might suggest avoiding making this a subclass. If we wanted, we could reuse behaviors of `ButtonComponent` by rendering it in the template: `render ButtonComponent.new(...)` instead of `content_tag(:button, ...)`

Reply:
Yeah the only thing I was extending was making the button get bigger but I easily remedied that by just adding 

` classes << 'usa-button--big' if big`

on line 25